src/code-util.jl

```diff
@@ -237,11 +237,14 @@ end
 
 matches_problem(prob1::Union{Tuple, Pair}, prob2) = matches_problem(Dict(prob1), prob2)
 matches_problem(prob1, prob2::Union{Tuple, Pair}) = matches_problem(prob1, Dict(prob2))
+
+# Match problems based on :code (required) and :line (if present in both)
```

Comment:
If the result has no line number and one is expected, this will pass which would be unexpected - but I don't think this can occur.